src/pages/learn/subscriptions.mdx

```diff
@@ -0,0 +1,102 @@
+As with query and mutation operations, GraphQL doesn't specify what transport protocol to use, so it's up to the server to decide. In practice, you will often see them implemented with WebSockets or server-sent events. Clients that want to send subscription operations will also need to support the chosen protocol. There are community-maintained specifications for implementing GraphQL subscriptions with [WebSockets](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) and [server-sent events](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md).
+
+One important distinction to make with subscription operations is that a document may contain a number of different subscription operations with different root fields, but each operation must have exactly one root field. 
+
+For example, this operation would be invalid:
+
+```graphql
+subscription {
+  reviewCreated {
+    rating
+    commentary
+  }
+  humanFriendsUpdated {
+    name
+    friends {
+      name
+    }
+  }
+}
+```
+
+But this operation would run:
+
+```graphql
+subscription NewReviewCreated {
+  reviewCreated {
+    rating
+    commentary
+  }
+}
+subscription FriendListUpdated($id: ID!) {
+  humanFriendsUpdated(id: $id) {
+    name
+    friends {
+      name
+    }
+  }
+}
+```
+
+As with query and mutation operations, the document above contains multiple subscription operations so each operation must be named.
+
+## Using subscriptions at scale
+
+Subscription operations are a powerful feature of GraphQL, but they are more complicated to implement than queries or mutations because you must maintain the GraphQL document, variables, and other context over the lifetime of the subscription. These requirements can be challenging when scaling GraphQL servers horizontally because each subscribed client must be bound to a specific instance of the server. 
+
+Similarly, client libraries will typically need more advanced features to handle these kinds of operations, such as logic to resubscribe to operations if the connection is disrupted or mechanisms to handle potential race conditions between initial queries and updated subscription data.
+
+In practice, a GraphQL API that supports subscriptions will call for a more complicated architecture than one that only exposes query and mutation fields. How this architecture is designed will depend on the specific GraphQL implementation, what pub/sub system supports the subscriptions, and the chosen transport protocol, as well as other requirements related to availability, scalability, and security of the real-time data.
+
+Subscription operations are best suited for data that changes often and incrementally, and for clients that need to receive those incremental updates as close to real-time as possible to deliver the expected user experience. For less frequent updates, periodic polling, mobile push notifications, or re-fetching queries based on user interaction may be the better solutions for keeping a client UI up to date.
```

Comment:
The text talked about subscriptions relating to "data that changes", but subscriptions are really well suited for events such as "Bob is typing..." or "Updated GraphQL schema available" that may not actually have a representation in the data and thus polling may not even be an option. I've tweaked this wording so that it doesn't exclude the pure-event use case:

```suggestion
Subscription operations are well suited for data that changes often and incrementally, and for clients that need to receive those incremental updates as close to real-time as possible to deliver the expected user experience. For data with less frequent updates, periodic polling, mobile push notifications, or re-fetching queries based on user interaction may be the better solutions for keeping a client UI up to date.
```